### PR TITLE
[FIX] stock: add automated way to fix unreserve issue

### PR DESCRIPTION
**Pull request by DavidFesquet**
When there is a discrepancy of reserved quantity between `stock.quant.reserved_quantity` and `stock.move.line.product_qty` the UserError 'It is not possible to unreserve more products of %s than you have in stock' is raised.
This discrepancy can be caused by past bug, customization or direct manipulation of `stock.quant` or `stock.move.line` by the users. Once this discrepancy is present, it will not disappear by itself.

Currently, there is no functional method to fix these discrepancies, the user encountering this message is stuck until they contact the Odoo support team.

Ideally, we would fix every bug related to the reservation so that this UserError almost never appears. However, even after multiple bugs fixed (ex: #144176 , #154327 ) and multiple years of feedback, the users keep encountering this error and keep getting stuck when validating/cancelling transfers.

So the idea is to re-introduce a server action like it was done for Odoo 13 & Odoo 14 2 years ago: d99e173b67ec2cfb99d057b5b8431344443a10a8 to the current stable version (except the latest one Odoo 17 as we do not have enough perspectives on this version).

The server action checks all the quants and their relative move line to check if match correctly. If not, it will remove the reservation from both. It could remove the reservation from some unrelated `pickings` and `stock.move`

Co-authored-by: Arnold Moyaux <arm@odoo.com>

Forward-Port of #79180
Related to #62139

---

Demo:

https://github.com/odoo/odoo/assets/29302288/1bc5fa9a-b0f6-4d37-b4f0-bb1e5d888420

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by DavidFesquet** [comment]
The error message can be adjusted.

As for the server action, it comes from this one: https://github.com/odoo/support-tools/blob/master/scripts/stock/fix_reservation_errors.py (support-tools repository, you may not have access to it)
We have been using it for more than a year now, and it works nicely).
A few adjustments have been made:
- simplified the report message
- remove the configuration
- remove the version dependent code

**Comment by DavidFesquet** [comment]
@fw-bot up to saas-15.2

File: addons/stock/models/stock_quant.py [hunk 1]
```diff
@@ -725,6 +738,7 @@ def _update_reserved_quantity(self, product_id, location_id, quantity, lot_id=No
             # if we want to unreserve
             available_quantity = sum(quants.mapped('reserved_quantity'))
             if float_compare(abs(quantity), available_quantity, precision_rounding=rounding) > 0:
+                self._raise_fix_unreserve_action(product_id)
                 raise UserError(_('It is not possible to unreserve more products of %s than you have in stock.', product_id.display_name))
```
Comment (amoyaux) [inline addons/stock/models/stock_quant.py:742]: You could add that the administrator has a way to correct it

**Comment by amoyaux** [comment]
@robodoo delegate+

**Comment by DavidFesquet** [comment]
robodoo r+

**Pull request closed without merging**